### ISSUE TITLE
Multiple `@Todo`s of the same type aren't considered equivelant

### ISSUE THREAD

**Issue by MeGaGiGaGon** [astral-sh/ty#3881]
### Summary

Since both `@Todo`s stem from the same generic in the same situation, I'd expect them to be the same, and thus be able to combine.
https://play.ty.dev/a66ddcc8-4bc7-4aee-89c9-3db405468b28
```py
import ty_extensions
from ty_extensions import TypeOf, static_assert, is_equivalent_to
from typing import reveal_type, Any

def doesnt_work[*TS](x: tuple[*TS], y: tuple[*TS]):
    z: TypeOf[x] & ~TypeOf[y] = 1  # ty:ignore[invalid-assignment, experimental-syntax]
    # Expected type: `Never`
    # Revealed type: `tuple[@Todo(TypeVarTuple), ...] & ~tuple[@Todo(TypeVarTuple), ...]` 
    reveal_type(z)
```
I'm aware of the idea of "different `Any`s could materialize to different types so can't be combined", but since both `@Todo`s in this case stem from the same generic, they should always be the same type, and should work with the normal type rules.

This comes from this minimization in my parser generators:
https://play.ty.dev/1ff00c51-1eee-4823-84e7-1ffc9b0950d1
Where this causes a similar `A & ~A` situation (that in my actual code causes an incorrect return type error)

Though I'm not sure this is worth/possible to do anything about, since my code would also work once `TypeVarTuple` support arrives.

### Version

playground e4945bde0

**Comment by sharkdp** [comment]
Thank you for reporting this. This rule could also apply to other dynamic types like `Any` or `Unknown` (in cases where you could *theoretically* infer that they have the exact same source), but we don't track the provenance of a type in this way. So as you suspected, this doesn't seem worth investing time in. Let's rather just wait for `TypeVarTuple` support / elimination of all `@Todo` types.
